Source/bmalloc/bmalloc/bmalloc.h

```diff
@@ -144,6 +144,34 @@ BINLINE void* memalign(size_t alignment, size_t size, CompactAllocationMode mode
 #endif
 }
 
+// Returns null on failure.
+BINLINE void* tryZeroedMemalign(size_t alignment, size_t size, CompactAllocationMode mode, HeapKind kind = HeapKind::Primary)
+{
+#if BUSE(LIBPAS)
+    if (!isGigacage(kind))
+        return bmalloc_try_allocate_zeroed_with_alignment_inline(size, alignment, asPasAllocationMode(mode));
+    return bmalloc_try_allocate_auxiliary_zeroed_with_alignment_inline(
+        &heapForKind(gigacageKind(kind)), size, alignment, asPasAllocationMode(mode));
+#else
+    BUNUSED(mode);
+    return Cache::tryAllocate(kind, alignment, size);
```

Comment:
Based on the implementation of `tryZeroedMalloc`, it looks like `Cache::tryAllocate` does not already zero the buffer.  You'll need to add zero'ing logic here.